OOP/LibraryManagement/LibraryApp/Services/IService.cs

```diff
@@ -0,0 +1,14 @@
+﻿using LibraryApp.Documents;
+
+namespace LibraryApp.Services;
+
+public interface IService<T> where T : Document
+{
+    T GetDocumentByTitle(string title);
+
+    IList<T> GetAllDocuments();
```

Comment:
From my point of view it' a good practise to follow the next rule: `accept more abstract argument, return more specific`. You can read about it [there](https://enterprisecraftsmanship.com/posts/return-the-most-specific-type/).  

In simple words - it depends on your case, however, if you are working in scope of app boundary context, then you should keep following this rule. If you are working with some library for export consumed by other programmers, then you can keep return types abstract.